config/apply/ccsm/translate.js

```diff
@@ -435,53 +424,59 @@ export function translateResponse(customApiResponse, patientData) {
       conclusionCodes = mapResults(excisionResults, customExcisionCodes, standardHistologyCodes, conclusionCodes);
     }
 
-    if (diagnosticReportIndex !== -1) {
-      console.log('Found the diagnostic report reference');
-      // Update the DiagnosticReport in patientData with the results from the custom API
-      patientData[diagnosticReportIndex] = {
-        ...patientData[diagnosticReportIndex],
-        code: {
-          coding: [
-            ...patientData[diagnosticReportIndex].code.coding,
-            ...codings
-          ]
-        },
-        conclusionCode: conclusionCodes
-      };
-
-      console.log('diagnostic report: ', patientData[diagnosticReportIndex]);
-      console.log('dr conclusion codes: ', patientData[diagnosticReportIndex].conclusionCode);
-      patientData[diagnosticReportIndex].conclusionCode.forEach(drcc => {
-        console.log('dr mapped conconclusion code: ', drcc.coding[0], drcc.text);
-      });
+    let patient = patientData.find(pd => pd.resourceType === 'Patient');
 
-      // Create a Procedure resource based on DiagnosticReport
-      if (procedureCoding) {
-        const originalDiagnosticReport = patientData[diagnosticReportIndex];
-        let newProcedure =
+    const laboratoryCategoryCode = {
+      'coding': [
         {
-          'resourceType': 'Procedure',
-          'id': originalDiagnosticReport.id,
-          'subject': originalDiagnosticReport.subject,
-          'status': 'completed',
-          'code': {
-            'coding': [procedureCoding],
-            'text': procedureText
-          },
-          'performedDateTime': originalDiagnosticReport.effectiveDateTime
-        };
-
-        if (newProcedure.id.length > 54) {
-          newProcedure.id = newProcedure.id.substring(0, 54) + '-procedure';
-        } else {
-          newProcedure.id += '-procedure';
+          'system': 'http://terminology.hl7.org/CodeSystem/v2-0074',
+          'code': 'LAB',
+          'display': 'Laboratory'
         }
+      ],
+      'text': 'Laboratory'
+    };
 
-        patientData.push(newProcedure);
+    // Create a DiagnosticReport resource from Order
```

Comment:
I think we should only create the new DiagnosticReport if the `codings` array is greater than 0, so we aren't automatically adding empty DRs to the patient record if there aren't actually any lab tests that we care about coming from customAPI? I just reviewed some of the testing results and for tests that don't have a relevant lab result we are automatically adding a diagnosticReport resource right now with an empty code.coding and conclusionCode.coding field.

Reply:
Nice idea to also skip the conclusionCode mapping if no `codings` exist ([referring to this code here](https://github.com/ccsm-cds-tools/ccsm-cds-service/pull/27/commits/1e99515f3c50f01dc13a1a5c4342f2abcaa507d2#diff-578021ab2cd5452e245d346c76b9c3d078fd8a4667b826589179302721e58eabR399-R403)), since a DiagnosticReport with a relevant `conclusionCode` but no relevant `code` would not be recognized by our CDS.

That being said, aren't there still other things later on in this function that we would need to execute, regardless of whether or not `codings` exist? Namely, I believe we would still want to.
1) Create a Procedure, since there can be instances where there is an "Excision" or an "Ablation", and we don't care about the result/DiagnosticReport, but our CDS still looks to check whether a "treatment" Procedure was done.
2) Translate EpisodeOfCare, which should not rely on `codings` at all.
3) Return the new `patientData` object if any of the above 2 are modified.

Reply:
Can CQL get a correct result with Procedure but without DiagnosticReport? I remember DiagnosticReport is must have and Procedure is optional.

Reply:
Yes, not every `order` from the customAPI needs to map to a DiagnosticReport for the CDS to generate a correct result. A good example is the excision results: we only actually care about converting the excision results to a DiagnosticReport if the result is AIS or histologic cancer ([see code here](https://github.com/ccsm-cds-tools/ccsm-cds-service/blob/80-build-dr/config/apply/ccsm/translate.js#L395)).  If the result of the excision is anything else, we don't need to generate a DiagnosticReport, we just need to generate a procedure, so our CDS knows that an excision was done, since there are a number of recommendations that rely on doing something after an excision (procedure) was done, regardless of what the "result" of the excision was.

Reply:
This was addressed by commit https://github.com/ccsm-cds-tools/ccsm-cds-service/pull/27/commits/1e99515f3c50f01dc13a1a5c4342f2abcaa507d2. The new organization of the steps in tranlate.js looks good. Now the Procedures and Episodes of care can be mapped within the `translateResponse()` independently of whether the DiagnosticReport `codings` exist.